### ISSUE TITLE
Segmentation Fault during sniff with Scapy development build on Ubuntu 18.04

### ISSUE THREAD

**Issue by grace-grimwood** [pynetwork/pypcap#84]
### Issue

Forgive me if this is better suited as an issue for the Scapy package, but I think the issue may be Pypcap so I thought I'd start here. I wrote a script to sniff on a given interface for a set period using Scapy's pcapdnet and Pypcap, but it seems that when Scapy attempts to close the socket (and therefore Pypcap tries to close the socket), a segmentation fault occurs. The same happens under the same conditions when using the Scapy CLI.

### Steps to reproduce

1. Install the latest release of Scapy via PIP
2. Install the latest version of Pypcap via PIP
3. Open the Scapy CLI as sudo and run the following code:
```
conf.use_pcap = True
import scapy.arch.pcapdnet
sniff(timeout=<insert arbitrary amount here>)
```

A segmentation fault should occur when the socket is being closed.

The same will happen if the following is run (as sudo) as a standalone python script:
```
from scapy.all import *
conf.use_pcap = True
import scapy.arch.pcapdnet
sniff(timeout=<insert arbitrary amount here>)
```
**NB:** Scapy has to be run as sudo on Ubuntu when using Pypcap, or an OSError is raised.

### System Specifications

  - Version: 1.2.2 (Scapy development build - secdev/scapy@94c71807c8e0eb729b34b421dd7726aae522391e)
  - Python Version: 2.7.15rc1
  - Platform: Ubuntu 18.04 (Kernel 4.15.0-42-generic)

### Images

I attempted to debug with gdb, but as I am very much not confident with Cython, much less C, I was having trouble making heads or tails of it. Here's a screencap of the gdb output.

![full gdb segfault](https://user-images.githubusercontent.com/15976368/49487721-9acb2980-f8a8-11e8-9a05-7a0403def6af.png)

**Comment by grace-grimwood** [comment]
Sorry, I've just checked and I seem to be running the development version of Scapy. I will try this again with the 2.4.0 release, and update if it is still an issue.

**Comment by grace-grimwood** [comment]
So the issue is resolved by switching to Scapy 2.4.0. It seems that it is the dev build of Scapy that is the issue, not Pypcap.

**Comment by grace-grimwood** [comment]
So I've pretty much confirmed this is an issue with the Scapy dev build, not Pypcap. Sorry if I wasted anyone's time with this!
